### ISSUE TITLE
`enum.member` loses callable value signature inside `Enum`

### ISSUE THREAD

**Issue by adamtheturtle** [astral-sh/ty#4573]
### Summary

`ty` loses the parameter signature of a callable wrapped with `enum.member` inside an `Enum`, producing a false-positive `invalid-assignment` diagnostic.

Playground: https://play.ty.dev/3cd50625-f074-4b9a-acc0-1dc0b345e552

```python
import enum
from collections.abc import Callable


def format_bytes(value: bytes, /) -> str:
    return value.hex()


class BytesFormats(enum.Enum):
    _value_: Callable[[bytes], str]

    HEX = enum.member(value=format_bytes)

    def __call__(self, data: bytes, /) -> str:
        return self._value_(data)
```

Command:

```console
$ ty check --python-version 3.12 repro.py
```

Actual result:

```text
error[invalid-assignment]: Enum member `HEX` value is not assignable to expected type
  --> repro.py:12:5
   |
12 |     HEX = enum.member(value=format_bytes)
   |     ^^^
info: Expected `(bytes, /) -> str`, got `() -> str`

Found 1 diagnostic
```

Expected result: no diagnostic. The wrapped value has type `Callable[[bytes], str]`, matching `_value_`, and the code is valid at runtime.

Pyright 1.1.414 and mypy 1.18.2 both accept this example. Explicitly specializing the typeshed generic under `TYPE_CHECKING` also makes `ty` accept it:

```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    BytesEnumMember = enum.member[Callable[[bytes], str]]
else:
    BytesEnumMember = enum.member
```

This suggests that the callable type is available from typeshed but is lost while `ty` processes `enum.member` as an enum member value.

Related: #1194 discusses ambiguous function-valued enum attributes and recommends wrapping them with `enum.member`; this report concerns the callable signature being lost after applying that wrapper.

### Version boundary

The reproducer passes without an error on `ty` 0.0.1 through 0.0.18 and starts producing the false-positive `invalid-assignment` diagnostic in 0.0.19. It continues to fail through 0.0.82.

The underlying inference problem predates the diagnostic: on 0.0.18,

```python
reveal_type(BytesFormats.HEX.value)
```

reveals `() -> str`, rather than `(bytes, /) -> str`. Version 0.0.19 added support for validating enum `_value_` annotations in astral-sh/ruff#22228, which exposed the existing incorrect inference as an error.

### Version

`ty 0.0.82 (6999edb88 2026-09-17)`

**Comment by carljm** [comment]
The root cause here is #3981 -- it's about our "callables binding as methods" heuristic. When we specialize `enum.member` with a "function-like callable" it means that `member.value` binds as a method, losing the first parameter. This would be fixed by https://github.com/astral-sh/ruff/pull/26778 or by https://github.com/astral-sh/ruff/pull/28428.
